src/workers/newNotificationMail.ts

```diff
@@ -0,0 +1,267 @@
+      post_views: post.views?.toLocaleString() ?? 0,
+      post_views_total: stats.numPostViews?.toLocaleString() ?? 0,
+      post_upvotes: post.upvotes?.toLocaleString() ?? 0,
+      post_upvotes_total: stats.numPostUpvotes?.toLocaleString() ?? 0,
+      post_comments: post.comments?.toLocaleString() ?? 0,
+      profile_link: addUtm(user.permalink, notification.type),
+    };
+  },
+  source_approved: async (con, user, notification, attachments, avatars) => {
+    const av = avatars[0];
+    const sourceRequest = await con
+      .getRepository(SourceRequest)
+      .findOneBy({ id: notification.referenceId });
+    return {
+      source_name: av.name,
+      source_image: av.image,
+      rss_link: sourceRequest.sourceFeed,
+      source_link: addUtm(notification.targetUrl, notification.type),
+    };
+  },
+  source_rejected: async (con, user, notification) => {
+    const sourceRequest = await con
+      .getRepository(SourceRequest)
+      .findOneBy({ id: notification.referenceId });
+    return {
+      first_name: getFirstName(user.name),
+      rss_link: sourceRequest.sourceFeed,
+    };
+  },
+  comment_mention: async (con, user, notification) => {
+    const comment = await con.getRepository(Comment).findOne({
+      where: { id: notification.referenceId },
+      relations: ['post', 'user'],
+    });
+    if (!comment) {
+      return;
+    }
+    const [commenter, post] = await Promise.all([comment.user, comment.post]);
+    return {
+      full_name: commenter.name,
+      commenter_profile_image: commenter.image,
+      comment: notification.description,
+      post_image: post.image || pickImageUrl(post),
+      post_title: truncatePostToTweet(post),
+      post_link: addUtm(notification.targetUrl, notification.type),
+      user_reputation: commenter.reputation,
+    };
+  },
+  comment_reply: async (con, user, notification) => {
+    const comment = await con.getRepository(Comment).findOne({
+      where: { id: notification.referenceId },
+      relations: ['parent', 'user', 'post'],
+    });
+    if (!comment) {
+      return;
+    }
+    const [commenter, parent, post] = await Promise.all([
+      comment.user,
+      comment.parent,
+      comment.post,
+    ]);
+    if (!commenter || !parent || !post) {
+      return;
+    }
+    return {
+      user_profile_image: user.image,
+      full_name: commenter.name,
+      main_comment: simplifyComment(parent.content),
+      new_comment: notification.description,
+      post_title: truncatePostToTweet(post),
+      discussion_link: addUtm(notification.targetUrl, notification.type),
+      user_reputation: user.reputation,
+      commenter_reputation: commenter.reputation,
+      commenter_profile_image: commenter.image,
+      user_name: user.name,
+    };
+  },
+  comment_upvote_milestone: async (con, user, notification) => {
+    return {
+      // Strip HTML tags
+      upvote_title: notification.title.replace(/<[^>]*>?/gm, ''),
```

Comment:
I'm aware that this is not a fully secured regex, but since we're in control of the notification title I don't think it's a problem. There will never be a `<script>` tag inside the title. I think we can ignore it.

Reply:
Firs time I'm even seeing this bot 🤯